onnxruntime/core/session/inference_session.cc

```diff
@@ -546,6 +546,48 @@ common::Status InferenceSession::InitializeSubgraphSessions(Graph& graph, Sessio
   return Status::OK();
 }
 
+static void ModelUseFP16Helper(const onnx::TypeProto type_proto, bool& use_fp16) {
```

Comment:
And `const onnx::TypeProto type_proto` should probably be taken by reference instead of by copy